### PR TITLE
zabbix packages - update to 3.0

### PR DESCRIPTION
**Pull request by dbaio**
zabbix packages - update to 3.0
Upgrade for zabbix agent and proxy packages
~~Pull TLS commit (cherry-pick) from master~~
Change web gui to TLS related options

### CODE REVIEW

**Comment by dbaio** [comment]
dependency: https://github.com/pfsense/pfsense/pull/2898

**Comment by rbgarga** [comment]
@dbaio can you please rebase your fork and remove changes to net-mgmt/zabbiz* ports? Keep only changes to pfSense specific ports please

**Comment by dbaio** [comment]
Hi @rbgarga,  It's done. Thanks.

File: net-mgmt/pfSense-pkg-zabbix-agent/files/usr/local/pkg/zabbix-agent-lts.inc [hunk 1]
```diff
@@ -250,4 +298,14 @@ EOF;
 	conf_mount_ro();
 }
 
+// Based on openvpn_add_keyfile() function
+function zabbix_agent_add_keyfile(& $data, $directive) {
```
Comment (rbgarga) [inline net-mgmt/pfSense-pkg-zabbix-agent/files/usr/local/pkg/zabbix-agent-lts.inc]: Why is $data a reference here if it's not modified inside the function?

File: net-mgmt/pfSense-pkg-zabbix-proxy/files/usr/local/pkg/zabbix-proxy-lts.inc [hunk 2]
```diff
@@ -218,4 +265,14 @@ EOF;
 	conf_mount_ro();
 }
 
+// Based on openvpn_add_keyfile() function
+function zabbix_proxy_add_keyfile(& $data, $directive) {
```
Comment (rbgarga) [inline net-mgmt/pfSense-pkg-zabbix-proxy/files/usr/local/pkg/zabbix-proxy-lts.inc]: Same here

File: net-mgmt/pfSense-pkg-zabbix-agent/files/usr/local/pkg/zabbix-agent-lts.inc (hunk 1 above)
Comment (dbaio) [inline net-mgmt/pfSense-pkg-zabbix-agent/files/usr/local/pkg/zabbix-agent-lts.inc]: copy/paste sorry

**Comment by MontgomeryTechnologies** [comment]
Hi Guys,

We desperately need to get the Zabbix Proxy working so we can forward SNMP traps back to a Zabbix server in a Colo.  I've engaged the original package writer, but he doesn't seem interested.  I've got $1000 that I'm willing to throw at this, if it will help. network at montgomerytech dot net if you'd like to claim the prize.  Thanks in advance and happy new year!

CT

**Comment by dbaio** [comment]
@MontgomeryTechnologies I am not interest on a bounty.

What version of zabbix (server) are you running?

This package is following zabbix 3.0 LTS 
http://www.zabbix.com/img/cycle/zabbix-standard-and-zabbix-lts-release-life-cycle_v3.2.png

Unfortunately zabbix proxy and server needs to be on the same version to work properly.

And I think is better you open a thread on [pfSense forum](https://forum.pfsense.org/index.php?board=15.0).

Regards.

**Comment by MontgomeryTechnologies** [comment]
Danilo,

Thanks for the response!

We are running Zabbix version 3.0.3 The version of proxy package and zabbix
are compatible.  Everything is working except SNMP traps being sent to
zabbix from pfsense proxy.

Make sense?

FYI

[image: Inline image 1]
[image: Inline image 2]



Eli Fogel

NOC Engineer
[image: cid:image001.jpg@01CEEF54.D25B4050]

Technology Manager for Premier Commercial Buildings Nationally

*Office **(844) 824.0100 <%28844%29%20824.0100> *| *Direct 415.918.2312
<+415.918.2312>* | efogel <http://goog_1187986303>*@montgomerytech.net
<http://montgomerytech.net>*

**Comment by MontgomeryTechnologies** [comment]
1)  Right now on the Zabbix-Proxy-LTS package that PFSense uses there is a
service that is being used to send snmp data called 'bsnmp'

2)  This service will not pass device snmp traps (port 162) through the
proxy package loaded on PFSense to the zabbix server.  All other snmp data
is being sent.

3)  Zabbix support has verified that this 'bsnmp' service will not do the
functions required for traps to be sent through the proxy agent.

4)  The service that needs to be used is 'snmptrapd' which is part of
net-snmp tools.

Is there any way that this service 'snmptrapd' could be loaded when the
package is loaded onto a PFSense box?  That way the snmp traps generated
from local network devices can be sent through the proxy agent to the
zabbix server?

Hope this explains it.  Please let me know if you have any other questions.

Thank you so much!


Eli Fogel

NOC Engineer
[image: cid:image001.jpg@01CEEF54.D25B4050]

Technology Manager for Premier Commercial Buildings Nationally

*Office **(844) 824.0100 <%28844%29%20824.0100> *| *Direct 415.918.2312
<+415.918.2312>* | efogel <http://goog_1187986303>*@montgomerytech.net
<http://montgomerytech.net>*

**Comment by MontgomeryTechnologies** [comment]
I have had a chance to test 2.4 Beta on a test PFSense box in our lab. When I first installed the Beta version of software we were not getting any traps. But the snmptrapd service was running on pfsense box and verified. After working with Zabbix and Danilo for the last month we have configured the PFSense proxy to pass traps to Zabbix in real time. Port status, coldstart, authentication etc.

Amongst a few config hurdles we discovered that everything on PFSense was configured correctly. A little more digging proved that the settings in the SNMP Trap GUI in PFSense were not being applied. The community names in the GUI were being overwritten by snmptrapd.conf file on PFSense. In the file the default public community is being applied. As soon as we modified this file to match our community name traps starting passing.

My summary and feedback is as follows:

The SNMP Trap GUI settings for community should be applied to the snmptrapd.conf file. Instead of going into the file manually, all changes made in the GUI should be final and applied across the board.

Thanks for reading. And thank you Danilo for all of your efforts and hard work.

**Pull request merged**